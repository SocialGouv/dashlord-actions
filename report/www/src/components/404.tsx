import * as React from "react";

import Table from "@codegouvfr/react-dsfr/Table";

import { Panel } from "./Panel";

type Report404Props = { data: Wget404Report };

export const Report404: React.FC<Report404Props> = ({ data }) => {
  const tableData = data && [
    ["destination"],
    ...data.map((url2) => [url2.link]),
  ];
  return (
    (data && data.length && (
      <Panel title="Erreurs 404" info="Pages introuvables">
        <Table data={tableData} />
      </Panel>
    )) ||
    null
  );
};
