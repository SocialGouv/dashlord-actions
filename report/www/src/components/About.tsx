import * as React from "react";

import CallOut from "@codegouvfr/react-dsfr/CallOut";

export const About: React.FC = () => (
  <CallOut className="fr-mb-3w" title="À propos">
    DashLord est <strong>100% open source</strong>, vous pouvez y contribuer en
    créant de la documentation, en proposant du code, ou encore en partageant
    vos questions{" "}
    <a
      href="https://github.com/socialgouv/dashlord"
      target="_blank"
      rel="noopener noreferrer"
    >
      sur le repository github
    </a>
    .
    <br /> <br />
    L&apos;objectif de DashLord est d&apos;identifier des points
    d&apos;attention mais aussi de proposer des solutions actionnables.
    <br /> <br />
    DashLord est né à la{" "}
    <a
      href="https://fabrique.social.gouv.fr"
      target="_blank"
      rel="noopener noreferrer"
    >
      {" "}
      Fabrique des ministères sociaux
    </a>{" "}
    pour répondre aux besoins d&apos;évaluation et de mise en oeuvre des bonnes
    pratiques de développement web.
    <br /> <br />
    Vous pouvez facilement créer votre propre DashLord pour vos URLs en suivant{" "}
    <a
      href="https://github.com/SocialGouv/dashlord/blob/main/README.md"
      target="_blank"
      rel="noopener noreferrer"
    >
      les instructions sur le GitHub
    </a>
    .
  </CallOut>
);
