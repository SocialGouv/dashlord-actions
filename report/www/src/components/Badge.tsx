import classNames from "classnames";
import React, { CSSProperties } from "react";
import { useRouter } from "next/router";

import styles from "./badge.module.scss";

type BadgeProps = {
  variant: string;
  className?: string;
  to?: string;
  style?: CSSProperties;
  external?: boolean;
  children: any;
};

const Badge: React.FC<BadgeProps> = ({
  children,
  external = false,
  variant,
  className,
  to,
  style,
}) => {
  const router = useRouter();
  return (
    <button
      type="button"
      className={classNames(className, styles[variant])}
      onClick={() => {
        if (to) {
          if (external) {
            window.open(to);
            return;
          }
          router.push(to);
        }
      }}
      style={style}
    >
      {children}
    </button>
  );
};

export default Badge;
