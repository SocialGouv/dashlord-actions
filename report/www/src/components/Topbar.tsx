import * as React from "react";
import { Home, Info, BarChart, Tool, LifeBuoy, GitHub, Search } from "react-feather";
import { Nav, Navbar, NavDropdown } from "react-bootstrap";
import { NavLink } from "react-router-dom";
import { smallUrl, sortByKey } from "../utils";

type TopbarProps = {
  report: DashLordReport;
};

export const Topbar: React.FC<TopbarProps> = ({ report }) => {
  const sortedReport = (report && report.sort(sortByKey("url"))) || [];
  return (<Navbar
    variant="dark"
    bg="dark"
    sticky="top"
    expand="lg"
    className="p-0 flex-md-nowrap shadow"
    style={{ justifyContent: 'space-between' }}
    collapseOnSelect
  >
    <Navbar.Brand href="#"><Home size={16} style={{ marginTop: -5, marginRight: 10 }} />DashLord</Navbar.Brand>
    <Navbar.Toggle aria-controls="responsive-navbar-nav" />
    <Navbar.Collapse id="responsive-navbar-nav">
      <Nav
        className="mr-auto my-2 my-lg-0"
        style={{ maxHeight: '100px' }}
        navbarScroll
      >
        <NavDropdown title="Agrégats" id="agregats">
          <NavDropdown.Item as={NavLink} to="/wappalyzer"><Tool size={16} style={{ marginTop: -5, marginRight: 10 }} />Technologies</NavDropdown.Item>
          <NavDropdown.Item as={NavLink} to="/trends"><BarChart size={16} style={{ marginTop: -5, marginRight: 10 }} />Évolutions</NavDropdown.Item>
        </NavDropdown>
        <NavDropdown title="Détails" id="details">
          {sortedReport.map((url) => (
            <NavDropdown.Item key={url.url} as={NavLink} to={`/url/${url.url}`}><Search size={16} style={{ marginTop: -5, marginRight: 5 }} />{smallUrl(url.url)}</NavDropdown.Item>
          ))}
        </NavDropdown>
      </Nav>
      <Nav>
        <Nav.Link as={NavLink} to="/intro"><LifeBuoy size={16} style={{ marginTop: -5, marginRight: 10 }} />Aide</Nav.Link>
        <Nav.Link as={NavLink} to="/about"><Info size={16} style={{ marginTop: -5, marginRight: 10 }} />À propos</Nav.Link>
        <Nav.Link href="https://github.com/SocialGouv/dashlord"><GitHub size={16} style={{ marginTop: -5, marginRight: 10 }} />Source</Nav.Link>
      </Nav>
    </Navbar.Collapse>
  </Navbar>
  )
};
