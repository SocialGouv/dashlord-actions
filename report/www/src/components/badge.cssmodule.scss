.success,
.info,
.warning,
.danger {
    display: inline-block;
    cursor: pointer;
    padding: 4px 8px;
    color: white;
    min-width: 60px;
    border-radius: 4px;
    text-align: center;
    font-weight: bold;
}

.success {
    background-color: var(--success);
}

.info {
    background-color: var(--info);
}

.warning {
    background-color: #FF6F4C;
}

.danger {
    background-color: var(--error);
}