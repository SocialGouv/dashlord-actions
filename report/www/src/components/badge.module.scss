.success,
.info,
.warning,
.danger {
    display: inline-block;
    cursor: pointer;
    padding: 4px 8px;
    color: white;
    min-width: 60px;
    border-radius: 4px;
    text-align: center;
    font-weight: bold;
}



.success, .success:hover {
    background-color: var(--success) !important;
}

.info, .info:hover {
    background-color: var(--info) !important;
}

.warning, .warning:hover {
    background-color: var(--warning) !important;
}

.danger, .danger:hover {
    background-color: var(--error) !important;
}