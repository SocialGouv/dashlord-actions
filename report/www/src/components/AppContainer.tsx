import * as React from "react";

import { Container, Alert } from "@dataesr/react-dsfr";

import { Switch, Route, useLocation, useParams } from "react-router-dom";

import { Dashboard } from "./Dashboard";
import { Trends } from "./Trends";
import { Intro } from "./Intro";
import { About } from "./About";
import { WappalyzerDashboard } from "./WappalyzerDashboard";
import Url from "./Url";

const report: DashLordReport = require("./report.json");
const trends: Trends = require("./trends.json");

type CategoryRouteProps = { report: DashLordReport };

// for some reason react-router `:url*` didnt work, use `*` only
interface CategoryParamTypes {
  category: string;
}

const CategoryRoute: React.FC<CategoryRouteProps> = (props) => {
  const params = useParams<CategoryParamTypes>();
  const category = window.decodeURIComponent(params.category);
  const urls = props.report.filter((u) => u.category === category);
  return (
    <>
      <br />
      {urls.length ? (
        <>
          <h3>
            {category} :{urls.length} urls
          </h3>
          <Dashboard report={urls} />
        </>
      ) : (
        <h3>Aucun URL associ√©e</h3>
      )}
    </>
  );
};

interface TagParamTypes {
  tag: string;
}

type TagRouteProps = { report: DashLordReport };

const TagRoute: React.FC<TagRouteProps> = (props) => {
  const params = useParams<TagParamTypes>();
  const tag = window.decodeURIComponent(params.tag);
  const urls = props.report.filter((u) => u.tags && u.tags.includes(tag));
  return (
    <>
      <br />
      <h3>
        {tag} :{urls.length} urls
      </h3>
      <Dashboard report={urls} />
    </>
  );
};

interface UrlParamTypes {
  "0": string;
}

type UrlRouteProps = { report: DashLordReport };

const UrlRoute: React.FC<UrlRouteProps> = (props) => {
  const params = useParams<UrlParamTypes>();
  const url = window.decodeURIComponent(params["0"]);
  const urlData = props.report.find((u) => u.url === url);
  if (!urlData) {
    return (
      <Alert
        type="error"
        title={`Impossible de trouver le rapport pour ${url}`}
      />
    );
  }
  return <Url url={url} report={urlData} />;
};

export const AppContainer = () => {
  const location = useLocation();
  return (
    <Container fluid={location.pathname === "/"}>
      <div role="main" className="fr-my-4w">
        <Switch>
          <Route path="/url/*">
            <UrlRoute report={report} />
          </Route>
          <Route path="/dashboard">
            <Dashboard report={report} />
          </Route>
          <Route path="/trends">
            <Trends trends={trends} />
          </Route>
          <Route path="/category/:category">
            <CategoryRoute report={report} />
          </Route>
          <Route path="/tag/:tag">
            <TagRoute report={report} />
          </Route>
          <Route path="/wappalyzer">
            <WappalyzerDashboard report={report} />
          </Route>
          <Route path="/intro">
            <Intro />
          </Route>
          <Route path="/about">
            <About />
          </Route>
          <Route path="/">
            <Dashboard report={report} />
          </Route>
        </Switch>
      </div>
    </Container>
  );
};
