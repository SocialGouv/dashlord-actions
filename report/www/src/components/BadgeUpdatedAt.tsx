import { formatDistanceToNow, differenceInDays, format } from "date-fns";
import { fr } from "@codegouvfr/react-dsfr";
import frLocale from "date-fns/locale/fr";

import { AlertProps } from "@codegouvfr/react-dsfr/Alert";
import Badge from "@codegouvfr/react-dsfr/Badge";
import { CSSProperties } from "react";

// badge that displya "since xxx" and appropriate colors
export const BadgeUpdatedAt = ({
  date,
  style,
}: {
  date: string;
  style: CSSProperties;
}) => {
  if (!date) return;
  const startDate = new Date(date);
  const lagInDays = differenceInDays(new Date(), startDate);
  let severity: AlertProps.Severity = "info";
  if (lagInDays > 60) {
    severity = "error";
  } else if (lagInDays > 30) {
    severity = "warning";
  }
  return (
    <Badge
      severity={severity}
      noIcon
      as="span"
      style={{ verticalAlign: "top", ...style }}
    >
      <i
        className={fr.cx("fr-icon-time-fill", "fr-icon--sm", "fr-mr-1v")}
        title={`Mis Ã  jour le ${format(new Date(startDate), "dd/M/yyyy")}`}
      />
      {formatDistanceToNow(new Date(startDate), {
        locale: frLocale,
      })}
    </Badge>
  );
};
