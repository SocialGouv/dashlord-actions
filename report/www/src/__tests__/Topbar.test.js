import * as React from "react";
import * as renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";

import { Topbar } from "../components/Topbar";

const report = require("../report.json");

it("Should render empty Topbar", () => {
  const props = {};
  const tree = renderer
    .create(
      <MemoryRouter>
        <Topbar {...props} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Should render full Topbar", () => {
  const props = { report };
  const tree = renderer
    .create(
      <MemoryRouter>
        <Topbar {...props} />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe("Tools config", () => {
  beforeEach(() => {
    jest.mock("../config.json", () => ({
      tools: ["http", "zap"],
    }));
  });

  it("Should render Topbar with limited tools", () => {
    const props = { report };
    const tree = renderer
      .create(
        <MemoryRouter>
          <Topbar {...props} />{" "}
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
