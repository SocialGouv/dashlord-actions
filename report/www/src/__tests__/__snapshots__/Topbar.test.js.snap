// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render empty Topbar 1`] = `
<nav
  className="p-0 flex-md-nowrap shadow navbar navbar-expand-lg navbar-light bg-light sticky-top"
  style={
    Object {
      "justifyContent": "space-between",
    }
  }
>
  <a
    className="navbar-brand"
    href="#"
  >
    <svg
      fill="none"
      height={16}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "marginRight": 10,
          "marginTop": -5,
        }
      }
      viewBox="0 0 24 24"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
      />
      <polyline
        points="9 22 9 12 15 12 15 22"
      />
    </svg>
    DashLord
  </a>
  <button
    aria-controls="responsive-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="responsive-navbar-nav"
  >
    <div
      className="mr-auto my-2 my-lg-0 navbar-nav navbar-nav-scroll"
      onKeyDown={[Function]}
      style={
        Object {
          "maxHeight": "100px",
        }
      }
    >
      <form
        className="form-inline"
      >
        <label
          className="form-label"
        >
          URL
        </label>
        <select
          className="form-control"
        >
          <option
            className=""
            href="/#"
            onClick={[Function]}
          >
            
          </option>
        </select>
      </form>
    </div>
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        aria-current={null}
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/wappalyzer"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height={16}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "marginRight": 10,
              "marginTop": -5,
            }
          }
          viewBox="0 0 24 24"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14.7 6.3a1 1 0 0 0 0 1.4l1.6 1.6a1 1 0 0 0 1.4 0l3.77-3.77a6 6 0 0 1-7.94 7.94l-6.91 6.91a2.12 2.12 0 0 1-3-3l6.91-6.91a6 6 0 0 1 7.94-7.94l-3.76 3.76z"
          />
        </svg>
        Technologies
      </a>
      <a
        aria-current={null}
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/trends"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height={16}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "marginRight": 10,
              "marginTop": -5,
            }
          }
          viewBox="0 0 24 24"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="12"
            x2="12"
            y1="20"
            y2="10"
          />
          <line
            x1="18"
            x2="18"
            y1="20"
            y2="4"
          />
          <line
            x1="6"
            x2="6"
            y1="20"
            y2="16"
          />
        </svg>
        Évolutions
      </a>
      <a
        aria-current={null}
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/intro"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height={16}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "marginRight": 10,
              "marginTop": -5,
            }
          }
          viewBox="0 0 24 24"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <circle
            cx="12"
            cy="12"
            r="4"
          />
          <line
            x1="4.93"
            x2="9.17"
            y1="4.93"
            y2="9.17"
          />
          <line
            x1="14.83"
            x2="19.07"
            y1="14.83"
            y2="19.07"
          />
          <line
            x1="14.83"
            x2="19.07"
            y1="9.17"
            y2="4.93"
          />
          <line
            x1="14.83"
            x2="18.36"
            y1="9.17"
            y2="5.64"
          />
          <line
            x1="4.93"
            x2="9.17"
            y1="19.07"
            y2="14.83"
          />
        </svg>
        Aide
      </a>
      <a
        aria-current={null}
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/about"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height={16}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "marginRight": 10,
              "marginTop": -5,
            }
          }
          viewBox="0 0 24 24"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <line
            x1="12"
            x2="12"
            y1="16"
            y2="12"
          />
          <line
            x1="12"
            x2="12.01"
            y1="8"
            y2="8"
          />
        </svg>
        À propos
      </a>
      <a
        className="nav-link"
        data-rb-event-key="https://github.com/SocialGouv/dashlord"
        href="https://github.com/SocialGouv/dashlord"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <svg
          fill="none"
          height={16}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "marginRight": 10,
              "marginTop": -5,
            }
          }
          viewBox="0 0 24 24"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
          />
        </svg>
        Source
      </a>
    </div>
  </div>
</nav>
`;

exports[`Should render full Topbar 1`] = `
<nav
  className="p-0 flex-md-nowrap shadow navbar navbar-expand-lg navbar-light bg-light sticky-top"
  style={
    Object {
      "justifyContent": "space-between",
    }
  }
>
  <a
    className="navbar-brand"
    href="#"
  >
    <svg
      fill="none"
      height={16}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "marginRight": 10,
          "marginTop": -5,
        }
      }
      viewBox="0 0 24 24"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
      />
      <polyline
        points="9 22 9 12 15 12 15 22"
      />
    </svg>
    DashLord
  </a>
  <button
    aria-controls="responsive-navbar-nav"
    aria-label="Toggle navigation"
    className="navbar-toggler collapsed"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="navbar-collapse collapse"
    id="responsive-navbar-nav"
  >
    <div
      className="mr-auto my-2 my-lg-0 navbar-nav navbar-nav-scroll"
      onKeyDown={[Function]}
      style={
        Object {
          "maxHeight": "100px",
        }
      }
    >
      <form
        className="form-inline"
      >
        <label
          className="form-label"
        >
          URL
        </label>
        <select
          className="form-control"
        >
          <option
            className=""
            href="/#"
            onClick={[Function]}
          >
            
          </option>
          <option
            className=""
            href="/url/https://adoption.gouv.fr"
            onClick={[Function]}
          >
            adoption.gouv.fr
          </option>
          <option
            className=""
            href="/url/https://alternance.emploi.gouv.fr"
            onClick={[Function]}
          >
            alternance.emploi.gouv.fr
          </option>
          <option
            className=""
            href="/url/https://arretonslesviolences.gouv.fr"
            onClick={[Function]}
          >
            arretonslesviolences.gouv.fr
          </option>
          <option
            className=""
            href="/url/https://bonusmalus.travail-emploi.gouv.fr"
            onClick={[Function]}
          >
            bonusmalus.travail-emploi.gouv.fr
          </option>
          <option
            className=""
            href="/url/https://boussole.jeunes.gouv.fr"
            onClick={[Function]}
          >
            boussole.jeunes.gouv.fr
          </option>
        </select>
      </form>
    </div>
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <a
        aria-current={null}
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/wappalyzer"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height={16}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "marginRight": 10,
              "marginTop": -5,
            }
          }
          viewBox="0 0 24 24"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14.7 6.3a1 1 0 0 0 0 1.4l1.6 1.6a1 1 0 0 0 1.4 0l3.77-3.77a6 6 0 0 1-7.94 7.94l-6.91 6.91a2.12 2.12 0 0 1-3-3l6.91-6.91a6 6 0 0 1 7.94-7.94l-3.76 3.76z"
          />
        </svg>
        Technologies
      </a>
      <a
        aria-current={null}
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/trends"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height={16}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "marginRight": 10,
              "marginTop": -5,
            }
          }
          viewBox="0 0 24 24"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="12"
            x2="12"
            y1="20"
            y2="10"
          />
          <line
            x1="18"
            x2="18"
            y1="20"
            y2="4"
          />
          <line
            x1="6"
            x2="6"
            y1="20"
            y2="16"
          />
        </svg>
        Évolutions
      </a>
      <a
        aria-current={null}
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/intro"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height={16}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "marginRight": 10,
              "marginTop": -5,
            }
          }
          viewBox="0 0 24 24"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <circle
            cx="12"
            cy="12"
            r="4"
          />
          <line
            x1="4.93"
            x2="9.17"
            y1="4.93"
            y2="9.17"
          />
          <line
            x1="14.83"
            x2="19.07"
            y1="14.83"
            y2="19.07"
          />
          <line
            x1="14.83"
            x2="19.07"
            y1="9.17"
            y2="4.93"
          />
          <line
            x1="14.83"
            x2="18.36"
            y1="9.17"
            y2="5.64"
          />
          <line
            x1="4.93"
            x2="9.17"
            y1="19.07"
            y2="14.83"
          />
        </svg>
        Aide
      </a>
      <a
        aria-current={null}
        className="nav-link"
        data-rb-event-key={null}
        disabled={false}
        href="/about"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height={16}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "marginRight": 10,
              "marginTop": -5,
            }
          }
          viewBox="0 0 24 24"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <line
            x1="12"
            x2="12"
            y1="16"
            y2="12"
          />
          <line
            x1="12"
            x2="12.01"
            y1="8"
            y2="8"
          />
        </svg>
        À propos
      </a>
      <a
        className="nav-link"
        data-rb-event-key="https://github.com/SocialGouv/dashlord"
        href="https://github.com/SocialGouv/dashlord"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <svg
          fill="none"
          height={16}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "marginRight": 10,
              "marginTop": -5,
            }
          }
          viewBox="0 0 24 24"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
          />
        </svg>
        Source
      </a>
    </div>
  </div>
</nav>
`;

exports[`Tools config Should render Topbar with limited tools 1`] = `
Array [
  <nav
    className="p-0 flex-md-nowrap shadow navbar navbar-expand-lg navbar-light bg-light sticky-top"
    style={
      Object {
        "justifyContent": "space-between",
      }
    }
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <svg
        fill="none"
        height={16}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "marginRight": 10,
            "marginTop": -5,
          }
        }
        viewBox="0 0 24 24"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
        />
        <polyline
          points="9 22 9 12 15 12 15 22"
        />
      </svg>
      DashLord
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="mr-auto my-2 my-lg-0 navbar-nav navbar-nav-scroll"
        onKeyDown={[Function]}
        style={
          Object {
            "maxHeight": "100px",
          }
        }
      >
        <form
          className="form-inline"
        >
          <label
            className="form-label"
          >
            URL
          </label>
          <select
            className="form-control"
          >
            <option
              className=""
              href="/#"
              onClick={[Function]}
            >
              
            </option>
            <option
              className=""
              href="/url/https://adoption.gouv.fr"
              onClick={[Function]}
            >
              adoption.gouv.fr
            </option>
            <option
              className=""
              href="/url/https://alternance.emploi.gouv.fr"
              onClick={[Function]}
            >
              alternance.emploi.gouv.fr
            </option>
            <option
              className=""
              href="/url/https://arretonslesviolences.gouv.fr"
              onClick={[Function]}
            >
              arretonslesviolences.gouv.fr
            </option>
            <option
              className=""
              href="/url/https://bonusmalus.travail-emploi.gouv.fr"
              onClick={[Function]}
            >
              bonusmalus.travail-emploi.gouv.fr
            </option>
            <option
              className=""
              href="/url/https://boussole.jeunes.gouv.fr"
              onClick={[Function]}
            >
              boussole.jeunes.gouv.fr
            </option>
          </select>
        </form>
      </div>
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          aria-current={null}
          className="nav-link"
          data-rb-event-key={null}
          disabled={false}
          href="/wappalyzer"
          onClick={[Function]}
        >
          <svg
            fill="none"
            height={16}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "marginRight": 10,
                "marginTop": -5,
              }
            }
            viewBox="0 0 24 24"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.7 6.3a1 1 0 0 0 0 1.4l1.6 1.6a1 1 0 0 0 1.4 0l3.77-3.77a6 6 0 0 1-7.94 7.94l-6.91 6.91a2.12 2.12 0 0 1-3-3l6.91-6.91a6 6 0 0 1 7.94-7.94l-3.76 3.76z"
            />
          </svg>
          Technologies
        </a>
        <a
          aria-current={null}
          className="nav-link"
          data-rb-event-key={null}
          disabled={false}
          href="/trends"
          onClick={[Function]}
        >
          <svg
            fill="none"
            height={16}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "marginRight": 10,
                "marginTop": -5,
              }
            }
            viewBox="0 0 24 24"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="20"
              y2="10"
            />
            <line
              x1="18"
              x2="18"
              y1="20"
              y2="4"
            />
            <line
              x1="6"
              x2="6"
              y1="20"
              y2="16"
            />
          </svg>
          Évolutions
        </a>
        <a
          aria-current={null}
          className="nav-link"
          data-rb-event-key={null}
          disabled={false}
          href="/intro"
          onClick={[Function]}
        >
          <svg
            fill="none"
            height={16}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "marginRight": 10,
                "marginTop": -5,
              }
            }
            viewBox="0 0 24 24"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <circle
              cx="12"
              cy="12"
              r="4"
            />
            <line
              x1="4.93"
              x2="9.17"
              y1="4.93"
              y2="9.17"
            />
            <line
              x1="14.83"
              x2="19.07"
              y1="14.83"
              y2="19.07"
            />
            <line
              x1="14.83"
              x2="19.07"
              y1="9.17"
              y2="4.93"
            />
            <line
              x1="14.83"
              x2="18.36"
              y1="9.17"
              y2="5.64"
            />
            <line
              x1="4.93"
              x2="9.17"
              y1="19.07"
              y2="14.83"
            />
          </svg>
          Aide
        </a>
        <a
          aria-current={null}
          className="nav-link"
          data-rb-event-key={null}
          disabled={false}
          href="/about"
          onClick={[Function]}
        >
          <svg
            fill="none"
            height={16}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "marginRight": 10,
                "marginTop": -5,
              }
            }
            viewBox="0 0 24 24"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="12"
              x2="12"
              y1="16"
              y2="12"
            />
            <line
              x1="12"
              x2="12.01"
              y1="8"
              y2="8"
            />
          </svg>
          À propos
        </a>
        <a
          className="nav-link"
          data-rb-event-key="https://github.com/SocialGouv/dashlord"
          href="https://github.com/SocialGouv/dashlord"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <svg
            fill="none"
            height={16}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "marginRight": 10,
                "marginTop": -5,
              }
            }
            viewBox="0 0 24 24"
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
            />
          </svg>
          Source
        </a>
      </div>
    </div>
  </nav>,
  " ",
]
`;
